core/services/ocr2/plugins/threshold/decryption_queue.go

```diff
@@ -0,0 +1,236 @@
+package threshold
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+type CiphertextId = []byte
+
+type ThresholdDecryptor interface {
+	Decrypt(ctx context.Context, ciphertextId CiphertextId, ciphertext []byte) ([]byte, error)
+}
+
+// This interface will be replaced by the Threshold Decryption Plugin when it is ready
+type DecryptionQueuingService interface {
+	GetRequests(requestCountLimit int, totalBytesLimit int) []DecryptionRequest
+	GetCiphertext(ciphertextId CiphertextId) ([]byte, error)
+	ReturnResult(ciphertextId CiphertextId, plaintext []byte)
```

Comment:
I'm late to the discussion here but my preference would be "SetResult" to follow a getter/setter naming pattern. Not important.

Reply:
I'll leave it for now as I believe Chrysa has already updated the name in her code too.